pages/update-profile.js

```diff
@@ -0,0 +1,102 @@
+import { useUser } from '@/utils/context/user'
+import { useState, useEffect } from 'react'
+import { useRouter } from 'next/router'
+import { supabase } from '@/utils/supabase'
+
+const trueFalse = [
+  { id: 1, text: 'True', value: true },
+  { id: 2, text: 'False', value: false }
+]
+
+export default function UpdateProfile() {
+  const { activeUser } = useUser()
+  const [updateData, setUpdateData] = useState({
+    avatar: '',
+    location: '',
+    isPasswordProtected: '',
+    pagePassword: '',
+    sites: ''
+  })
```

Comment:
Instead of manually adding your object keys, could you just set the `activeUser` (simplify this to `user`) here? Then you could remove that `useEffect`

```jsx
const [updateData, setUpdateData] = useState(user)
```

Reply:
The issue I was having with that, was that if there was no activeUser, I kept getting the "error reading undefined" message, so I had to use the useEffect.